packages/core-components/src/layout/SignInPage/GuestUserIdentity.ts

```diff
@@ -19,14 +19,44 @@ import {
   ProfileInfo,
   BackstageUserIdentity,
 } from '@backstage/core-plugin-api';
+import { Config } from '@backstage/config';
+import nJwt from 'njwt';
+import { GUEST_USER_PRIVATE_KEY } from '@backstage/plugin-auth-common';
 
 export class GuestUserIdentity implements IdentityApi {
+  private idToken?: string;
+
+  static fromConfig(config?: Config): GuestUserIdentity {
+    const newIdentity = new GuestUserIdentity();
+    if (config?.getOptionalBoolean(`auth.allowGuestMode`)) {
+      newIdentity.setIdToken(newIdentity.issueToken());
+    }
+    return newIdentity;
+  }
+
+  private setIdToken(token: string) {
+    this.idToken = token;
+  }
+
+  private issueToken(): string {
+    const userEntityRef = `user:default/guest`;
+    const idToken: string = nJwt
+      .create(
+        { sub: userEntityRef, ent: [userEntityRef], aud: 'backstage' },
```

Comment:
I think its definitely worth thinking about. This seemed a bit "out of scope" for this iteration in my opinion but it is something i looked at while working on this. The biggest hurdle I ran into was that on the backend TokenFactory uses the jose library to create a JWKS that can be used when creating the JWT. I tried to replicate this on the frontend but unfortunately i ran into an error ever time i tried to introduce a unit-test  because when frontend modules are tested, they run in a jsdom environment which lacks support for TextEncoder. Jose has a hard dependency on TextEncoder with no plans to change [this](https://github.com/panva/jose/issues/307), so a move in this direction might mean figuring out a different library than jose that can be tested on the frontend to provide us a token with the correct JWK.